### PR TITLE
Added gateway functionality for dahsboards

### PR DESCRIPTION
**Pull request by jtennyson724**
Added kxdash.q file which allows async queries while connected to the gateway via kx dashboards

### CODE REVIEW

File: code/gateway/kxdash.q [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+.gw.formatresponse:{[status;sync;result]
+ if[`kxdash~first result;
+  res:last result;
+  :$[res`status;
+    (`.dash.rcv_msg;res`w;res`o;res`r;res`result);
+    (`.dash.snd_err;res`w;res`r;res`result)]];
+ $[not[status]and sync;'result;result]}
+
+// incorporate dashps into the .z.ps definition
+.z.ps:dashps
```
Comment (jonnypress) [inline code/gateway/kxdash.q:52]: Change the definition of the formatresponse function to be in the .kxdash namespace, then add an init function which
a) sets .gw.formatresponse to have the definition from the .kxdash namespace
b) resets .z.ps

Add a variable .kxdash.enabled which defaults to false (same as the way TorQ handles other variables). If set to true, the init function is invoked. 

The definition of.z.ps should also probably be wrapped the way the other message handler definitions are wrapped, e.g. 

.z.ps:{x@y; .kxdash.dashps y}@[value;`.z.ps;{{value x}}]
The init function should be invoked if a

Short version is we probably don't want this to be implemented by default, but we want it to be easy to turn on.

**Comment by shehir94** [comment]
@jonnypress have pushed some updates here addressing comments. Have tested with @MorrisStranger-aq

**Review by jonnypress — APPROVED** [review]
